csharp/src/Microsoft.ML.OnnxRuntime/Training/TrainingSession.shared.cs

```diff
@@ -568,8 +568,10 @@ public void OptimizerStep(RunOptions options)
         /// an inference model if it knows the inference graph outputs. The input inference graph outputs
         /// are used to prune the eval model so that the inference model's outputs align with the provided outputs.
         /// The exported model is saved at the path provided and can be used for inferencing with InferenceSession.
-        /// Note that the function re-loads the eval model from the path provided to TrainingSession
-        /// and expects that this path still be valid.
+        ///
+        /// This function modifies the eval graph in-place, so after this method is called, the TrainingSession can
+        /// no longer be used for training. In order to continue training from this point, save the checkpoint state
+        /// and create a new TrainingSession with the saved checkpoint state.
```

Comment:
Out of interest is it better/possible to save the checkpoint first or does it need some changes that exportModelForInference makes to be included in it?

Reply:
It's possible to save the checkpoint first -- exporting for inference shouldn't affect the checkpoint / it won't matter in what order those functions are called